addons/account/models/account_move.py

```diff
@@ -3576,7 +3582,7 @@ class AccountMoveLine(models.Model):
     amount_residual_currency = fields.Monetary(string='Residual Amount in Currency', store=True,
         compute='_compute_amount_residual',
         help="The residual amount on a journal item expressed in its currency (possibly not the company currency).")
-    full_reconcile_id = fields.Many2one('account.full.reconcile', string="Matching", copy=False, index=True, readonly=True)
+    full_reconcile_id = fields.Many2one('account.full.reconcile', string="Matching", copy=False, index="not null", readonly=True)
```

Comment:
this is questionable, it looks tuned for people who don't use stock 
1. full reconcile id in a trading database with AngloSaxon will have a much lower percentage of nulls. Less than 50% typically.
2. A very common search is for unreconciled items. You can't just consider the percentage of Nulls, but rather how often people wish to search for NULL's and I'd argue it is more common than for a particular reconcile record. account_move_line is typically one of the largest tables and not using an index for a NULL search is nuts.